src/libAtomVM/term.h

```diff
@@ -1106,6 +1111,33 @@ static inline term term_create_empty_binary(size_t size, Heap *heap, GlobalConte
     return t;
 }
 
+static inline bool term_normalize_binary_pos_len(term binary, avm_int_t pos, avm_int_t len, BinaryPosLen *pos_len)
+{
+    avm_int_t size = (avm_int_t) term_binary_size(binary);
+    if (len < 0) {
+        pos += len;
+        len = -len;
+    }
+
+    if (UNLIKELY((pos < 0) || (pos > size) || (pos + len > size))) {
+        return false;
+    }
+
+    pos_len->pos = pos;
+    pos_len->len = len;
+    return true;
+}
+
+static inline bool term_is_invalid_binary_pos_len(BinaryPosLen pos_len)
+{
+    return pos_len.pos == -1 && pos_len.len == -1;
+}
+
+static inline BinaryPosLen term_invalid_binary_pos_len(void)
+{
+    return (BinaryPosLen) { .pos = -1, .len = -1 };
```

Comment:
C allows returning a struct (by value), but this make the compiler generating memory copy code.
Here you are using an inline function, so actually it might be optimized out, but I', not sure about this.
So if you want to initialize this struct in an always optimized manner let's use macros.

Reply:
One last thing, `(void)` is not necessary.

Reply:
I mostly want to be consistent with other code. If we want to avoid copies, we can use `__attribute__((always_inline))`. I think this should be done with everything in this file and only after benchmarking. For this PR, I'd avoid doing that.